HDSAviz/interactivevislib.py

```diff
@@ -0,0 +1,186 @@
+
+    Output:
+    Tab based plotting of Graphs.
+    """
+    working_dir = !pwd
+    if working_dir != 'C:/HDSAviz_data/' :
+        change_directory()
+
+    analysis_files = gather_all_analysis_files()
+    tabs_dictionary = {}
+
+    for i,files in zip(range(len(analysis_files)), analysis_files):
+        # cleaning the file names to get a short name to occupy less
+        # space in tabs.
+        names = files.replace('analysis_', '' ).replace('.txt','')
+
+        try :
+            tabs_dictionary[i] = Panel(child = makeplot(files,
+                                np.array(['S1', 'ST']),
+                                np.array([0.001*S1_Range, 0.001*ST_Range])),
+                                title = names)
+        except:
+            pass
+
+
+    tabs = Tabs(tabs=tabs_dictionary.values())
+
+    try :
+        p = show(tabs)
+        output_notebook()
+    except:
+        pass
+
+    return p
+
+def Interact_with_burtin_plots():
+    """
+    This function adds capability to adjust the minimum threshold for
+    sensitiity plotting
+
+    Output:
+    Interactive graph displaying a slider widgets that allow for
+    minimum threshold of sensitivity selection
+    """
+    output_notebook()
+    slider1 = FloatSlider(value = 1, min = 0.1, max = 10,
+                          step=0.1, title = 'S1-Range')
+    slider2 = FloatSlider(value = 1, min = 0.1, max = 10,
+                          step=0.1, title = 'ST-Range')
+    return interact(plot_all_outcomes_burtin, S1_Range=slider1,
+                    ST_Range=slider2)
+
+def short_tabs_demo(S1_Range,ST_Range):
+    """
+    This function plots the burtin type plots for two outcome
+    variables in CO and CO2.
+
+    Input :
+    S1_Range: Minimum value of sensitivity for first
+              order effects * 0.001,
+    ST_Range: Minimum value of sensitivity for total
+              order effects * 0.001
+
+    Output:
+    Tab based plotting of Graphs (Only two tabs for simplicity
+    and sanity check experimentation).
+
+    """
+    output_notebook()
+    tab1 = Panel(child = makeplot('analysis_CO.txt', np.array(['S1', 'ST']),
+                                  np.array([0.001*S1_Range, 0.001*ST_Range])),
+                                  title = 'CO')
+    tab2 = Panel(child = makeplot('analysis_CO2.txt', np.array(['S1', 'ST']),
+                                  np.array([0.001*S1_Range, 0.001*ST_Range])),
+                                  title = 'CO2')
+    tabs = Tabs(tabs=[tab1,tab2])
+    return show(tabs)
+
+def short_interactive_demo ():
+    """
+    This function adds capability to adjust the minimum threshold
+    for sensitiity plotting for demo and uses only two key factors.
+
+    Output:
+    Interactive graph displaying a slider widgets that allow for
+    minimum threshold of sensitivity selection
+    """
+    #output_notebook()
+    slider1 = FloatSlider(value = 1, min = 0.1, max = 10,
```

Comment:
I think we need to come up with a come up a consistent way to refer to the different variables to avoid confusion. (i.e. parameters refers to Tmax and C and factors refers to S1 or S2) We can talk about this in person though